sqlobject/src/main/java/org/jdbi/v3/sqlobject/customizer/BindList.java

```diff
@@ -55,20 +55,26 @@
      */
     enum EmptyHandling {
         /**
-         * output "" (without quotes, i.e. nothing)
-         * <p>
-         * select * from things where x in ()
+         * <p>Output "" (without quotes, i.e. nothing).</p>
+         *
+         * <code>select * from things where x in ()</code>
          */
         VOID,
         /**
-         * output "null" (without quotes, as keyword), useful e.g. in postgresql where "in ()" is invalid syntax
-         * <p>
-         * select * from things where x in (null)
+         * <p>Output "null" (without quotes, as keyword), useful e.g. in postgresql where "in ()" is invalid syntax.</p>
+         *
+         * <code>select * from things where x in (null)</code>
          */
         NULL,
         /**
-         * throw IllegalArgumentException
+         * Throw IllegalArgumentException.
          */
-        THROW;
+        THROW,
+        /**
+         * <p>Define an empty list, leaving the resulting query text up to the {@link org.jdbi.v3.core.statement.TemplateEngine} to decide.</p>
+         *
+         * This value was specifically added to <a href="https://github.com/jdbi/jdbi/issues/1377">make conditionals work better with <code>StringTemplate</code></a>.
+         */
+        EMPTY_LIST
```

Comment:
Out of the 3 options, it seems to be the clearest and most semantic: this is about BindList so we define a List value, skipping the definition is functionally misleading because we did high-level bind/define something so the Map should show that, and defining `null`... I dunno, it may work, I'll test it, but I don't like the way a null value entry corresponds in usage to the key not being present at all when a value check is done by the readers of the Map. It's misleading, you know? Is it null because it's not defined or because a null was put into it, you can't tell.

Reply:
Also, I had no idea what to name an option that defines `null` when we already called the `"null"` one `NULL` 🤷‍♂️ So since binding an empty list works just as well, and avoids the ever-dreaded `null`, I went with that...

Reply:
Just pushed a commit with this change. It works at least. I would prefer the empty list variant, but it's a soft preference. If you have a hard preference for the other one, I'll roll with it.

Reply:
I just didn't see any evidence that the empty list was actually ever used as anything except a "falsy" value.  If our goal is to insert a falsy value, I expect `null` is more widely accepted as falsy, `"[]" == null` could easily be a StringTemplate-ism.

Also, I agree that inserting a literal null is icky, we can simply omit defining it in that case.

Reply:
Yeah, it was intended as a StringTemplate-ism, hence the javadoc. We explicitly support StringTemplate, so I figured it ok.

I kinda don't wanna just omit the definition because it's still a null but now also only as a side-effect instead of explicit behavior.

Reply:
Well, `map.put(key, null)` is not generally safe.  Depending on the map implementation it will throw.  If you really want to do something we should probably `map.remove(key)` instead, as that should always have the desired effect.

Reply:
We'll change that when it happens then